nornir/core/inventory.py

```diff
@@ -105,6 +105,18 @@ def __contains__(self, value: object) -> bool:
         else:
             return any([value == g for g in self])
 
+    def add(self, group: "Group") -> None:
+        # only add the group if it doesn't exist
+        if not self.__contains__(group):
+            self.append(group)
+
+    def remove(self, group: "Group") -> None:
+        if self.__contains__(group):
+            # remove overrides the parent class so calling super()
+            super(ParentGroups, self).remove(group)
+        else:
+            raise KeyError(group)
```

Comment:
Not sure I like this check, if you try to remove something that doesn’t exist I think it should error out. In addition, in the standard library removing an element from a list raises a ValueError if it doesn’t exist but you are raising a KeyError. I think we should be consistent with the stdlib, in which case this method wouldn’t be needed.

Reply:
Good call.